### ISSUE TITLE
`void` as return type is not supported by mock generator

### ISSUE THREAD

**Issue by mageekguy** [atoum/atoum#710]
Given:

```
<?php namespace estvoyage\risingsun\comparison;

interface unary
{
	function recipientOfComparisonWithOperandIs($operand, recipient $recipient) :void;
}
```

Then `new \mock\estvoyage\risingsun\comparison\` generate the error:

```
==> Error FATAL ERROR in /Users/mageekguy/Risingsun/tests/units/src/comparison/unary/not.php on unknown line, generated by file /Users/mageekguy/Risingsun/vendor/atoum/atoum/classes/mock/generator.php(198) : eval()'d code on line 30:
A void function must not return a value
```

The dump of mock code is:

```
==> output(1796) "string(1780) "namespace mock\estvoyage\risingsun\comparison {
==> final class unary implements \estvoyage\risingsun\comparison\unary, \mageekguy\atoum\mock\aggregator
==> {
==> 	public function getMockController()
==> 	{
==> 		$mockController = \mageekguy\atoum\mock\controller::getForMock($this);
==> 		if ($mockController === null)
==> 		{
==> 			$this->setMockController($mockController = new \mageekguy\atoum\mock\controller());
==> 		}
==> 		return $mockController;
==> 	}
==> 	public function setMockController(\mageekguy\atoum\mock\controller $controller)
==> 	{
==> 		return $controller->control($this);
==> 	}
==> 	public function resetMockController()
==> 	{
==> 		\mageekguy\atoum\mock\controller::getForMock($this)->reset();
==> 		return $this;
==> 	}
==> 	public function recipientOfComparisonWithOperandIs($operand, \estvoyage\risingsun\comparison\recipient $recipient): void
==> 	{
==> 		$arguments = array_merge(array($operand, $recipient), array_slice(func_get_args(), 2));
==> 		if (isset($this->getMockController()->recipientOfComparisonWithOperandIs) === false)
==> 		{
==> 			$this->getMockController()->recipientOfComparisonWithOperandIs = function() {};
==> 		}
==> 		$return = $this->getMockController()->invoke('recipientOfComparisonWithOperandIs', $arguments);
==> 		return $return;
==> 	}
==> 	public function __construct(\mageekguy\atoum\mock\controller $mockController = null)
==> 	{
==> 		$this->{'mock' . uniqid()} = true;
==> 		if ($mockController === null)
==> 		{
==> 			$mockController = \mageekguy\atoum\mock\controller::get();
==> 		}
==> 		if ($mockController !== null)
==> 		{
==> 			$this->setMockController($mockController);
==> 		}
==> 		if (isset($this->getMockController()->__construct) === true)
==> 		{
==> 			$this->getMockController()->invoke('__construct', func_get_args());
==> 		}
==> 	}
==> 	public static function getMockedMethods()
==> 	{
==> 		return array (
==>   0 => 'recipientofcomparisonwithoperandis',
==>   1 => '__construct',
==> );
==> 	}
==> }
==> }""
```

Moreover:

```
$ php -v
PHP 7.1.3 (cli) (built: Apr 14 2017 08:55:03) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies
$ php tests/units/runner.php --version
atoum version dev-master
```

**Comment by jubianchi** [comment]
this will be fixed with #701. I'll try to merge today

**Comment by jubianchi** [comment]
#701 was merged.
